### ISSUE TITLE
[Task]: Cleanup / reorganize the transformation code

### ISSUE THREAD

**Issue by chouinar** [HHS/simpler-grants-gov#1926]
### Summary

As we built the transformation code quickly, there are a few parts of it (listed below) that we'd like to take another pass at for code-quality reasons. This is meant as a fast-follow or something we can do while testing the transformation code.

* Restructure how the files are laid out, right now there are two files that all transformation logic is in, consider separating it a bit more.
* Metrics - we just have some basic high-level counts, but metrics for specific tables would help a lot
* Boilerplate code - a lot of the transformation code follows similar patterns, see if we can trim down the boilerplate code a bit by making re-usable utility methods

We should timebox this work to complete while we're waiting on testing

### Acceptance criteria

- [ ] Clean up based on the bullets above

**Comment by chouinar** [comment]
Duplicate of https://github.com/HHS/simpler-grants-gov/issues/2054
